frontend/src/lib/v2/CompareUtils.ts

```diff
@@ -0,0 +1,139 @@
+/*
+ * Copyright 2022 The Kubeflow Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { CompareTableProps, xParentLabel } from 'src/components/CompareTable';
+import { getArtifactName, getExecutionName } from 'src/mlmd/MlmdUtils';
+import { getMetadataValue } from 'src/mlmd/Utils';
+import { ExecutionArtifact, RunArtifact } from 'src/pages/CompareV2';
+import { Value } from 'src/third_party/mlmd';
+import * as jspb from 'google-protobuf';
+import { chain, flatten } from 'lodash';
+
+// Add the scalar metric names and data items.
+const addScalarDataItems = (
+  customProperties: jspb.Map<string, Value>,
+  scalarMetricNames: string[],
+  dataMap: { [key: string]: string },
+  artifactIndex: number,
+) => {
+  for (const entry of customProperties.getEntryList()) {
+    const scalarMetricName: string = entry[0];
+    if (scalarMetricName === 'display_name') {
+      continue;
+    }
+    scalarMetricNames.push(scalarMetricName);
+
+    // The scalar metric name and artifact index create a unique key per data item.
+    const key: string = `${scalarMetricName}-${artifactIndex}`;
```

Comment:
Let's extract the following logic and this highlighted line to a common function because it is used in multiple places, in this way, changing one location won't cause mismatch in another location.

```
const key: string = `${yLabel}-${artifactIndex}`
```

Reply:
Sounds good, done here: https://github.com/kubeflow/pipelines/pull/8026/commits/d876504dc37aa6a6e935d38ecfdcb35c5da7d56f